apps/meteor/app/api/server/v1/users.ts

```diff
@@ -756,9 +782,9 @@ API.v1.addRoute('users.2fa.sendEmailCode', {
 		}
 
 		const method = emailOrUsername.includes('@') ? 'findOneByEmailAddress' : 'findOneByUsername';
-		const userId = this.userId || Users[method](emailOrUsername, { fields: { _id: 1 } })?._id;
+		const userId = this.userId || (await Users[method](emailOrUsername, { projection: { _id: 1 } }))?._id;
 
-		const user = await getUserForCheck(userId);
+		const user = await getUserForCheck(userId || '');
```

Comment:
dont use `SPACE/empty strings` to cheat on the user.username typings
there is a test exactly on the line below to prevent that